### PR TITLE
bitmex shift fetchOHLCV timestamps to emulate open timestamps 

### PR DESCRIPTION
**Pull request by frosty00**

### CODE REVIEW

**Comment by kroitor** [comment]
@frosty00 need to make sure that the number of candles returned is equal to `limit` and to verify that the first returned candle actually contains the correct data corresponding to their website. Also, might have to do `limit + 1` to get all results. Ideally, we would add this to https://github.com/ccxt/ccxt/blob/master/js/test/Exchange/test.fetchOHLCV.js

File: build/transpile.js [hunk 1]
```diff
@@ -191,8 +191,8 @@ const pythonRegexes = [
     [ /if\s+\((.*)\)\s*[\n]/g, "if $1:\n" ],
     [ /\}\s*else\s*\{/g, 'else:' ],
     [ /else\s*[\n]/g, "else:\n" ],
-    [ /for\s+\(([a-zA-Z0-9_]+)\s*=\s*([^\;\s]+\s*)\;[^\<\>\=]+(?:\<=|\>=|<|>)\s*(.*)\.length\s*\;[^\)]+\)\s*{/g, 'for $1 in range($2, len($3)):'],
-    [ /for\s+\(([a-zA-Z0-9_]+)\s*=\s*([^\;\s]+\s*)\;[^\<\>\=]+(?:\<=|\>=|<|>)\s*(.*)\s*\;[^\)]+\)\s*{/g, 'for $1 in range($2, $3):'],
+    [ /for\s+\(([a-zA-Z0-9_]+)\s*=\s*([^\;\s]+\s*)\;[^\<\>\=]+(?:\<=|\>=|<|>)\s*(.*)\.length\s*\;\s*\1+(-|\+){2}\)\s*{/g, 'for $1 in range($2, len($3), $41):'],
+    [ /for\s+\(([a-zA-Z0-9_]+)\s*=\s*([^\;\s]+\s*)\;[^\<\>\=]+(?:\<=|\>=|<|>)\s*(.*)\s*\;\s*\1+(-|\+){2}\)\s*{/g, 'for $1 in range($2, $3, $41):'],
```
Comment (kroitor) [inline build/transpile.js]: 41? Not 42? )

Comment (frosty00) [reply build/transpile.js]: its actually a $4 next to a 1

**Comment by brandsimon** [comment]
Shouldnt fetchOHLCVOpenTimestamp be true by default, since it is the ccxt default behavior?
I know, it could break someones code, but I dont see having inconsistent behavior as a solution.
Is there a way to communicate breaking changes to the community? Maybe over increasing the version number to 1.19 and putting this information in to the release info on github?

There was also a rename of markets somewhere, which I think should be properly communicated.

**Comment by kroitor** [comment]
> Shouldnt fetchOHLCVOpenTimestamp be true by default, since it is the ccxt default behavior?

Yes.

> Is there a way to communicate breaking changes to the community? 

We are trying to announce critical changes beforehand, but will try better.

> Maybe over increasing the version number to 1.19 and putting this information in to the release info on github?

Yep, we usually change the patch number (the third number). We switch the minor version number when the master goes diverse from the old version.

> Maybe over increasing the version number to 1.19 and putting this information in to the release info on github?

Unfortunately, the problem is that writing release info takes more time than writing the new functionality, fixing important bugs, and creating code itself...

In fact, we're already writing the release info, because the latest commit message goes to the tags automatically: 

Screen Shot 2019-07-16 at 18 15 36

We've been asked about changelogs previously here:

- https://github.com/ccxt/ccxt/issues/2483#issuecomment-379490600
- https://github.com/ccxt/ccxt/issues/76

... it's basically a tradeoff of _"useful work"_ vs _"bureaucracy"_.

> There was also a rename of markets somewhere, which I think should be properly communicated.

Totally agree with you on this, need to figure out a way to automate listing/delisting announcements, changes in unified symbols, etc...

**Comment by hroff-1902** [comment]
I think the approach implemented here
```
emulate the open timestamp by shifting all the timestamps one place
so the previous close becomes the current open, and we drop the first candle
```
is wrong. Just assume that data has gaps. You won't get proper open timestamps by shifting them. The open timestamps should be calculated in this case, by subtracting the bar duration from the close timestamp which is returned by BitMEX.

**Comment by kroitor** [comment]
@hroff-1902 thx for the comment, I think you're right, we will rework this a bit.

File: js/base/Exchange.js [hunk 2]
```diff
@@ -1615,4 +1615,19 @@ module.exports = class Exchange {
             throw new ExchangeError (this.id + ' this.twofa has not been set')
         }
     }
+
+    normalizeTimeframe (timeframe) {
```
Comment (kroitor) [inline js/base/Exchange.js]: What about parseTimeframe ? )

**Pull request merged**